Src/IronPython.Modules/grp.cs

```diff
@@ -0,0 +1,207 @@
+ * ironpy@microsoft.com. By using this source code in any fashion, you are agreeing to be bound 
+ * by the terms of the Microsoft Public License.
+ *
+ * You must not remove this notice, or any other, from this software.
+ *
+ *
+ * ***********************************************************************/
+
+#if FEATURE_NATIVE || NETCOREAPP2_0
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.IO;
+using System.IO.Pipes;
+using System.Runtime.InteropServices;
+
+using Microsoft.Scripting.Runtime;
+
+using IronPython.Runtime;
+using IronPython.Runtime.Operations;
+
+using System.Numerics;
+
+[assembly: PythonModule("grp", typeof(IronPython.Modules.PythonGrp), PythonModuleAttribute.PlatformFamily.Unix)]
+namespace IronPython.Modules {
+    
+    public static class PythonGrp {
+        public const string __doc__ = @"Access to the Unix group database.
+        
+Group entries are reported as 4-tuples containing the following fields
+from the group database, in order:
+
+  gr_name   - name of the group
+  gr_passwd - group password (encrypted); often empty
+  gr_gid    - numeric ID of the group
+  gr_mem    - list of members
+  
+The gid is an integer, name and password are strings.  (Note that most
+users are not explicitly listed as members of the groups they are in
+according to the password database.  Check both databases to get
+complete membership information.)";
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct group {
+            [MarshalAs(UnmanagedType.LPStr)]
+            public string gr_name;
+            [MarshalAs(UnmanagedType.LPStr)]
+            public string gr_passwd;
+            public int gr_gid;
+            public IntPtr gr_mem;
+        };
+
+        [PythonType("struct_group")]
+        [Documentation(@"grp.struct_group: Results from getgr*() routines.
+
+This object may be accessed either as a tuple of
+  (gr_name,gr_passwd,gr_gid,gr_mem)
+or via the object attributes as named in the above tuple.
+")]
+        public class struct_group : PythonTuple {
+
+            private const int LENGTH = 4;
+
+            internal struct_group(string gr_name, string gr_passwd, int gr_gid, List gr_mem) {
+                this.gr_name = gr_name;
+                this.gr_passwd = gr_passwd;
+                this.gr_gid = gr_gid;
+                this.gr_mem = gr_mem;
+            }
+
+            [Documentation("group name")]
+            public string gr_name { get; }
+
+            [Documentation("password")]
+            public string gr_passwd { get; }
+
+            [Documentation("group id")]
+            public int gr_gid { get; }
+
+            [Documentation("group members")]
+            public List gr_mem { get; }
+
+            public override int __len__() {
+                return LENGTH;
+            }
+
+            private object[] AsArray() {
+                return new object[] { gr_name, gr_passwd, gr_gid, gr_mem };
```

Comment:
Seems a bit inefficient to create an array every time we do a tuple operation? An option would be to initialize the tuple in the constructor and just make the properties index ops on the underlying array? That way we wouldn't need any of the overrides (other than `__repr__`).